ttnn/cpp/ttnn/operations/data_movement/reshape_view/reshape.cpp

```diff
@@ -48,172 +80,376 @@ ttnn::Tensor host_reshape(const ttnn::Tensor& tensor, const ttnn::Shape& shape)
+ttnn::Shape tiling_reshape_corrector(const ttnn::Shape& shape, const uint32_t tile_first_dim, const uint32_t tile_second_dim) {
+    //Apply the correct padding metadata to the target shape
     auto padded = shape.with_tile_padding();
     auto rank = shape.rank();
-    const int8_t correction_1 =
-        (ttnn::types::TILE_SIZE - (int)padded[-1] % ttnn::types::TILE_SIZE) % ttnn::types::TILE_SIZE;
-    if (rank == 1) {
-        return ttnn::Shape({1, shape[0]}, {32, padded[0] + correction_1});
-    }
-    const int8_t correction_2 =
-        (ttnn::types::TILE_SIZE - (int)padded[-2] % ttnn::types::TILE_SIZE) % ttnn::types::TILE_SIZE;
-    switch (rank) {
-        case 2: return ttnn::Shape({shape[0], shape[1]}, {padded[0] + correction_2, padded[1] + correction_1}); break;
+    const int8_t correction_1 =(tile_first_dim - (int)padded[-1] % tile_first_dim) % tile_first_dim;
+    if(rank == 1)
+    {
+        return ttnn::Shape({1,shape[0]},{32,padded[0]+correction_1});
+    }
+    const int8_t correction_2 =(tile_second_dim - (int)padded[-2] % tile_second_dim) % tile_second_dim;
+    switch(rank)
+    {
+        case 2:
+            return ttnn::Shape({shape[0],shape[1]},{padded[0]+correction_2,padded[1]+correction_1});
+            break;
         case 3:
-            return ttnn::Shape(
-                {shape[0], shape[1], shape[2]}, {padded[0], padded[1] + correction_2, padded[2] + correction_1});
+            return ttnn::Shape({shape[0],shape[1],shape[2]},{padded[0],padded[1]+correction_2,padded[2]+correction_1});
             break;
         case 4:
-            return ttnn::Shape(
-                {shape[0], shape[1], shape[2], shape[3]},
-                {padded[0], padded[1], padded[2] + correction_2, padded[3] + correction_1});
+            return ttnn::Shape({shape[0],shape[1],shape[2],shape[3]},{padded[0],padded[1],padded[2]+correction_2,padded[3]+correction_1});
             break;
+
     }
     return shape;
 }
 
-ttnn::Tensor PerformView(const ttnn::Tensor& tensor, const ttnn::Shape& shape) {
+ttnn::Tensor PerformView(const ttnn::Tensor& tensor, const ttnn::Shape& shape, const uint32_t tile_first_dim, const uint32_t tile_second_dim) {
+    if (tensor.get_shape() == shape) {
+        return tensor;
+    }
     if (tensor.get_layout() == ttnn::TILE_LAYOUT &&
-        (shape[-1] % ttnn::types::TILE_SIZE != 0 || shape[-2] % ttnn::types::TILE_SIZE != 0)) {
-        // Correct the output shape to add padding metadata before reshape (view)
-        return tensor.reshape(tiling_reshape_corrector(shape));
+        (shape[-1]%tile_first_dim!=0 || shape.rank()==1 || shape[-2]%tile_second_dim!=0 ))
+    {
+        //Correct the output shape to add padding metadata before reshape (view)
+        return tensor.reshape(tiling_reshape_corrector(shape, tile_first_dim, tile_second_dim));
     }
-    // Perform a reshape (view)
+    //Perform a reshape (view)
     return tensor.reshape(shape);
```

Comment:
How soon will we actually have a view and reshape call so its really clear when cost is guaranteed to be ~0

Reply:
There is a separate PR right now for metadata-only reshapes that will cause PCC issues if incorrectly used. However I like to consolidate the guidance to always use ttnn.reshape that does the view if safe to do so. Maybe I can add a can_do_view operation which will tell the model writer if it is going to be a view or not in advance? I have also provided clear guidance in the pybind to state when a 0 cost view is happening.